includes/class-indieauth-authenticate.php

```diff
@@ -164,23 +172,24 @@ public function verify_state( $state ) {
 	 */
 	public function authenticate( $user ) {
 		$redirect_to = array_key_exists( 'redirect_to', $_REQUEST ) ? $_REQUEST['redirect_to'] : null;
+		$redirect_to = rawurldecode( $redirect_to );
 		if ( array_key_exists( 'indieauth_identifier', $_POST ) && $_POST['indieauth_identifier'] ) {
 			$me = esc_url_raw( $_POST['indieauth_identifier'] );
 			// Check for valid URLs https://indieauth.spec.indieweb.org/#user-profile-url
 			if ( ! wp_http_validate_url( $me ) ) {
 				return new WP_Error( 'indieauth_invalid_url', __( 'Invalid User Profile URL', 'indieauth' ) );
 			}
-			$this->authorization_redirect( $me, $redirect_to );
+			$this->authorization_redirect( $me, wp_login_url( $redirect_to ) );
 		} elseif ( array_key_exists( 'code', $_REQUEST ) && array_key_exists( 'state', $_REQUEST ) ) {
 			$state = $this->verify_state( $_REQUEST['state'] );
 			if ( is_wp_error( $state ) ) {
 				return $state;
 			}
-			$me = $this->verify_authorization_token( $_REQUEST['code'], $redirect_to );
-			if ( is_wp_error( $me ) ) {
-				return $me;
+			$response = $this->verify_authorization_token( $_REQUEST['code'], wp_login_url( $redirect_to ) );
```

Comment:
just for consistency, you should probably call this function `verify_authorization_code`